platforms/software/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/transform/ArtifactTransformInputArtifactIntegrationTest.groovy

```diff
@@ -348,20 +361,29 @@ class ArtifactTransformInputArtifactIntegrationTest extends AbstractDependencyRe
         when:
         succeeds(":a:resolve")
 
-        then: // have already seen these artifacts before, but the transform outputs have been overwritten
+        then:
         result.assertTasksNotSkipped(":b:producer", ":a:resolve")
-        transformed("b-dir")
+        // have already seen these artifacts before...
+        if (incremental) {
+            // ...but the transform outputs have been overwritten
+            transformed("b-dir")
+        } else {
+            // ...and they are up-to-date
+            transformed()
+        }
         outputContains("result = [b-dir.green, c-dir.green]")
 
         where:
-        description                                 | annotation
-        "no sensitivity"                            | ""
-        "@PathSensitive(PathSensitivity.ABSOLUTE)"  | "@PathSensitive(PathSensitivity.ABSOLUTE)"
-        "@PathSensitive(PathSensitivity.RELATIVE)"  | "@PathSensitive(PathSensitivity.RELATIVE)"
-        "@PathSensitive(PathSensitivity.NAME_ONLY)" | "@PathSensitive(PathSensitivity.NAME_ONLY)"
+        [sensitivity, incremental] << [
```

Comment:
💅This looks a bit complicated, can it be simplified with a table?

Reply:
While I agree this is not the nicest API from Spock, we do use `.combinations()` in several other places in our tests to handle similar situations where we need to test all the combinations of two vectors. This format lets us discuss the two dimensions without introducing redundancy.

I tried doing the de-normalized table version, but I'm not sure I like it any better:

```groovy
where:
incremental | sensitivity
true        | null
true        | PathSensitivity.ABSOLUTE
true        | PathSensitivity.RELATIVE
true        | PathSensitivity.NAME_ONLY
false       | null
false       | PathSensitivity.ABSOLUTE
false       | PathSensitivity.RELATIVE
false       | PathSensitivity.NAME_ONLY
```

I don't feel strongly either way, so let's go with your suggestion.